163383135 Add Social Login Functionality

missvicki opened the pull request:
**Description**

The application needs to have some level of authentication in order to allow certain functionalities such as creating an article such as to determine what articles belong to which user which user has commented on article.

**Type of change**

- New feature

**How has this been tested**

- Unit tests testing individual functionalities

**Relevant Pivotal Tracker Stories** 

[#163383135](https://www.pivotaltracker.com/story/show/163383135)

fahdjamy reviewed (approved):
Thanks for the one on one explanation. 
@missvicki and @oma0256

codjoero commented on package.json:
```diff
@@ -3,25 +3,31 @@
+    "start": "HTTPS=true react-scripts start",
     "build": "react-scripts build",
-    "test": "react-scripts test",
+    "test": "react-scripts test -u",
```
Good job on this feature @missvicki  thanks for the one on one explanation. though it might not be a good idea to leave the `-u` flag on your test script. It will keep replacing your snapshots with new ones, hence eroding the whole point of snapshot testing.

missvicki replied:
Thank you for spotting this, I will make the correction.

codjoero reviewed (commented):
Great work @missvicki thanks for the one on one explanation. Responsiveness is good, but on mobile particularly at 462, the UI would look much better with a centered modal.

codjoero commented on src/config.json:
```diff
@@ -0,0 +1,4 @@
+{
+  "GOOGLE_CLIENT_ID": "134825392968-d9p3skmcrstu1q0t8ll6nhikphc8ovjg.apps.googleusercontent.com",
+  "FACEBOOK_CLIENT_ID": 229135374703025
+}
```
@missvicki  EOL

oma0256 commented:
> Great work @missvicki thanks for the one on one explanation. Responsiveness is good, but on mobile particularly at 462, the UI would look much better with a centered modal.

@codjoero about this @missvicki and I worked on the responsiveness on the other PR. This will be fixed once we get the other PR merged

codjoero reviewed (approved):
With the explanation, all is good

karuhanga reviewed (changes requested):
Good work on this @missvicki.
Just a few observations;
- Is it possible for you to either add a loading indicator- until it is fully active on the facebook login button or find a way of making the styling appear and the button active, because it is not working in some browsers, and I think that is important
- Could the background image cover the entire screen? @FahdJamy can probably help with this
![image](https://user-images.githubusercontent.com/12200860/54409617-d885ec80-46f8-11e9-9e94-3856e0613cb1.png)
(white spaces on the sides.)

missvicki commented:
> Good work on this @missvicki.
> Just a few observations;
> 
> * Is it possible for you to either add a loading indicator- until it is fully active on the facebook login button or find a way of making the styling appear and the button active, because it is not working in some browsers, and I think that is important
> * Could the background image cover the entire screen? @FahdJamy can probably help with this
>   ![image](https://user-images.githubusercontent.com/12200860/54409617-d885ec80-46f8-11e9-9e94-3856e0613cb1.png)
>   (white spaces on the sides.)

@Karuhanga i do not think a loading indicator is necessary here. the buttons are customized by google and facebook and set to open up a new window. We found difficulty in customizing the buttons to fit our preferences. 

We are working on making the background better.

karuhanga commented:
> > Good work on this @missvicki.
> > Just a few observations;
> > 
> > * Is it possible for you to either add a loading indicator- until it is fully active on the facebook login button or find a way of making the styling appear and the button active, because it is not working in some browsers, and I think that is important
> > * Could the background image cover the entire screen? @FahdJamy can probably help with this
> >   ![image](https://user-images.githubusercontent.com/12200860/54409617-d885ec80-46f8-11e9-9e94-3856e0613cb1.png)
> >   (white spaces on the sides.)
> 
> @Karuhanga i do not think a loading indicator is necessary here. the buttons are customized by google and facebook and set to open up a new window. We found difficulty in customizing the buttons to fit our preferences.
> 
> We are working on making the background better.

Agreed. I think I used the word `loading` incorrectly. What I meant was as the `css` is still loading for the `facebook` button, you could show that it is not active yet.
But on further inspection, it seems like it just doesn't load on `firefox`, rather than it taking long.

**UPDATE**
So the `facebook` login button does not work by default because `firefox` automatically blocks setting cookies by foreign domains. Unfortunately we have no control over this. So, we'll let that slide until it is a critical issue.

pbkabali commented:
@missvicki  @oma0256 , your explanations helped with this. Thanks for the good work.

karuhanga reviewed (approved):
Looks good now.
We''l need to look into improving the styling in the future though.

collinewait commented on src/components/Navbar.jsx:
```diff
@@ -62,11 +63,12 @@ export const Navbar = (props) => {
-Navbar.propTypes = {
-    navBar: PropTypes.object.isRequired
-};
+// Navbar.propTypes = {
```
Commented code

collinewait commented on src/store/actions/async/__tests__/login.test.js:
```diff
@@ -18,6 +19,7 @@ describe("login async actions", () => {
             .then(() => {
                 expect(store.getActions()).toEqual([
                     {"type": "LOGIN_STARTED"}, { userData: {user:{ username: "oma0256", "token": "sdbfh"}}, type: LOGIN_USER }
+                    // { userData: {user:{ username: "oma0256", "token": "sdbfh"}}, type: LOGIN_USER }
```
commented code.

collinewait commented on src/views/LoginView.jsx:
```diff
@@ -36,39 +41,100 @@ export class LoginView extends Component {
+    }
+    
+
+    onFailure = error => {
```
Is this method doing anything?
`no console logs`

collinewait commented on src/components/Navbar.jsx:
```diff
@@ -9,6 +9,7 @@ import branding from '../assets/images/branding.png';
 
 export const Navbar = (props) => {
     const { navBar } = props;
+    console.log(navBar)
```
Remove `console.log` statement.

oma0256 committed:
163383135-feature(login social user): implement functionality for social login to authors haven

    - allow a user view a login buttons for social facebook and google
    - user can click either social buttons and log in with either social platform
    - redirects user to home page on successful login.
    - write tests for the social login functionality.

[Starts #163383135]

Co-authored-by: Victor Nomwesigwa <nvbethany479@gmail.com>
